### PR TITLE
Canceling a single request

### PR DESCRIPTION
**Pull request by mejackreed**
- [x] Cancel a single request that is ready for pickup
 - [x] Change a pickup library that is not yet ready for pickup
 - [x] Change a not needed by date that is not yet ready for pickup
 - [x]  Figure out the correct api call for not needed by
 - [x] Figure out how to put what libraries can pickup be changed to
 - [x] Cancel a request that is not yet ready for pickup
 - [x] ~Put it in a modal~ Split to #242

change

### CODE REVIEW

File: app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -19,4 +19,65 @@ def index
                   end
                 end
   end
+
+  def edit
+    @request = patron.requests.select { |r| r.key == params['id'] }.first
```
Comment (cbeer) [inline app/controllers/requests_controller.rb]:
```suggestion
    @request = patron.requests.find { |r| r.key == params['id'] }
```

File: app/models/library_location.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#  Class to handle configuration and logic around library codes and labels
+#  Adapted from https://github.com/sul-dlss/sul-requests/blob/master/app/models/library_location.rb
+###
+class LibraryLocation
+  def initialize(home_location)
+    @home_location = home_location
+  end
+
+  def additional_pickup_libraries(current_pickup_library)
+    pickup_libraries.reject { |l| l == current_pickup_library }
```
Comment (cbeer) [inline app/models/library_location.rb]:
```suggestion
    pickup_libraries.except(current_pickup_library)
```

File: app/models/library_location.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  # rubocop:enable Style/EmptyCaseCondition
+
+  private
+
+  def library_specific_pickup_libraries?
+    config.library_specific_pickup_libraries.key?(@home_location)
+  end
+
+  def location_specific_pickup_libraries?
+    config.location_specific_pickup_libraries.key?(@home_location)
```
Comment (cbeer) [inline app/models/library_location.rb]: Maybe we should pull `config.location_specific_pickup_libraries` out to a private method so it's a little less unwieldy.

Comment (mejackreed) [reply app/models/library_location.rb]: Great recommendation!

**Comment by mejackreed** [comment]
@camillevilla oops looks like I had that commented out. I popped on 13d0f1d which should resolve the issue. I had it commented out before I knew about the morison way to create requests.

**Pull request merged**